public/app/features/dashboard/dashgrid/PanelChrome.tsx

```diff
@@ -144,6 +146,43 @@ export class PanelChrome extends PureComponent<Props, State> {
         console.log('No width yet... wait till we know');
         return;
       }
+      if (panel.datasource === SHARED_DASHBODARD_QUERY) {
+        const query = panel.targets[0] as DashboardQuery;
+        const otherPanel = this.props.dashboard.getPanelById(query.panelId);
+        if (otherPanel) {
+          if (otherPanel.queryRunner === panel.queryRunner) {
+            if (this.props.isFullscreen) {
+              otherPanel.refresh(); // TODO
+              console.log('Somehow refresh the other query when in fullscreen');
+            }
+            return; // should be good
+          }
+          if (this.querySubscription) {
+            this.querySubscription.unsubscribe();
+          }
+          if (!otherPanel.queryRunner) {
+            otherPanel.queryRunner = new PanelQueryRunner();
+          }
+          // Use the same runner?  HACK -- easy way to check if we are already listening
+          panel.queryRunner = otherPanel.queryRunner;
```

Comment:
I think it's a good idea, just wondering if it won't cause troubles when i.e. "other" panel is removed. IIRC the dashgrid is force updated on panel removal, so it should probably be fine.

Reply:
Also, the benefit is that if other panel issues queries from a panel that relies on other Dashboard query, then  you basically always use the original runner, having a single source of truth for results. And we can optimise number of request as well using single runner, so that multiple panels don't send requests for the same query. SGTM

Reply:
exactly!